python_modules/dagster/dagster/_core/definitions/declarative_automation/automation_condition.py

```diff
@@ -482,7 +482,7 @@ def on_cron(cron_schedule: str, cron_timezone: str = "UTC") -> "AutomationCondit
     @public
     @experimental
     @staticmethod
-    def on_missing() -> "AutomationCondition":
+    def on_missing() -> "AndAutomationCondition":
         """Returns an AutomationCondition which will cause missing asset partitions to be materialized as soon as possible,
         after all of their dependencies have been materialized.
 
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines +442 to +488]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->I don't think we should change these typehints<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24226?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
right now, only `AndAutomationCondition` supports this feature, so I wanted to make sure the type system could understand that `AutomationCondition.eager().without(...)` was valid